website/project/model.py

```diff
@@ -316,6 +352,12 @@ def edit(self, content, auth, save=False):
         self.content = content
         self.modified = True
         self.date_modified = datetime.datetime.utcnow()
+        new_mentions = get_valid_mentioned_users_guids(self, self.node.contributors)
+
+        if new_mentions:
+            project_signals.mention_added.send(self, new_mentions=new_mentions, auth=auth)
+            self.ever_mentioned.extend(new_mentions)
+
         if save:
```

Comment:
Not sure if this will answer your question but awhile ago I polled the [dev flow](https://www.flowdock.com/app/cos/osf-dev?filter=search&query=can%20anyone%20explain%20why%20many) with a similar question.